### PR TITLE
Bechurch/21720 versioned connector build status

### LINKED ISSUE #21720: QA Engine: retrieve connector build status
## Tell us about the problem you're trying to solve
Depends on https://github.com/airbytehq/airbyte/issues/20442
We want to know per connector and version if their latest build is successful.

## Describe the solution you’d like
We can parse the HTML results from `https://dnsgjos7lj2fu.cloudfront.net/tests/summary/{connector_technical_name}/index.html`
 
Sample from the spike (not per version):
```python
def latest_build_is_successful(status_page_url) -> bool:
    # TODO how to make sure the latest build matches the OSS catalog connector version?
    # Parse raw logs from S3?
    response = requests.get(status_page_url)
    if response.status_code != 200:
        return False
    page = response.text

    history = []
    for row in re.findall(r"<tr>(.*?)</tr>", page):
        cols = re.findall(r"<td>(.*?)</td>", row)
        if not cols or len(cols) != 3:
            continue
        history.append(
            {
                "date": cols[0],
                "status": re.findall(r" (\S+)</span>", cols[1])[0],
                "link": re.findall(r'href="(.*?)"', cols[2])[0],
            }
        )
    return history[0]["status"] == "success"
```

## Describe the alternative you’ve considered or used
Build reports are also stored on S3. It can give us access to the rawest data that could help define for which connector version the build was run.

### PR DESCRIPTION
**Pull request by bnchrch**
## What
Closes #21720

This issue needs the connector version alongside the build status for each connector.

This is so that we can proactively mark connectors as "Ready for cloud"

## How
This accomplishes it with the following changes
1. Retrieve the version from the dockerfile in the report.sh script
2. Use the version to output the test artifacts at `$CONNECTOR/$VERSION`
3. Consume those artifacts in the qa_engine to determine build status and version

## What else
We've made an additional effort in this PR to ensure that
1. All build logs for connectors are recorded to `tests/history/connectors` not `tests/history`. (Currently they are being recorded to both)
2. The relevant build statuses point to `tests/history/connectors`

## Recommended reading order
1. tools/status/report.sh:write_job_log
3. tools/status/report.sh:generate_job_log_json
4. tools/ci_connector_ops/ci_connector_ops/qa_engine/validations.py:latest_build_is_successful
5. tools/ci_connector_ops/ci_connector_ops/qa_engine/inputs.py:fetch_latest_build_status_for_connector_version
6. tools/bin/build_report.py

## 🚨 User Impact 🚨
No breaking change if successful.

However, there is a risk here that a part of the system is made stale. This is because we've made an effort to standardize the build report path, which means that theres a chance we missed a system that still references the old path.

Though thats low risk low impact

## Pre-merge Checklist
Expand the relevant checklist and delete the others.
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [x] Build is successful
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [x] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

## Post merge

- [ ] run build_report workflow
- [ ] run qa-engine workflow

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: tools/bin/build_report.py
Comment (bechurch) [inline tools/bin/build_report.py:33]: @sherifnada @evantahler I just wanted to make you aware of this path change. (And the related changes to `builds.md` and `report.sh`).

Essentially we had two paths that were in use for connector builds status
1. `tests/history`
2. `tests/history/connectors`

This change wrangles this to ensure we are always using and referencing `tests/history/connectors`

Let me know if theres a system impacted that I may have missed

**Comment by evantahler** [comment]
Screenshot 2023-01-30 at 1 23 04 PM
Github hack - if you use the phrase "closes #ticket" it will link up the PR and Issue, and close the issue when the PR is merged.

https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue

File: airbyte-integrations/builds.md
Comment (evantahler) [inline airbyte-integrations/builds.md:9]: Looks like these URLs don't resolve yet.

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/inputs.py
Comment (evantahler) [inline tools/ci_connector_ops/ci_connector_ops/qa_engine/inputs.py:27]: nit: From the test below, this would be the string "success" if all went well. Maybe this method should return a Boolean?  Or, perhaps be an ENUM type  for the returned string values?

File: tools/lib/lib.sh
Comment (evantahler) [inline tools/lib/lib.sh:45]: nice!

File: tools/status/report.sh
Comment (evantahler) [inline tools/status/report.sh:18]: 👍 I don't think a "/" would be in a connector name

**Review by evantahler — APPROVED** [review]
Nice work Ben!

File: tools/status/report.sh
Comment (bechurch) [reply tools/status/report.sh:18]: haha ya, i didn't feel the best about this but bash wise I think its good

File: tools/ci_connector_ops/ci_connector_ops/qa_engine/inputs.py
Comment (bechurch) [reply tools/ci_connector_ops/ci_connector_ops/qa_engine/inputs.py:27]: I like this idea.

Originally it was a bit awkward as the build outcome being a string was already used in a lot of placed.

I wanted to go and update that as well but it felt a bit out of scope.

So I opted in [validation.py](https://github.com/airbytehq/airbyte/pull/22029/files#diff-fdd65838aa29c584c0ea92de3e6b169a0e054d356828982b37f87c1ec804e501R39) to transform the string to a boolean.

Putting at least an enum in input and using it in validation is a good idea though. I will change.

File: airbyte-integrations/builds.md
Comment (bechurch) [reply airbyte-integrations/builds.md:9]: No they should resolve when merged to master. 

You can see an example here:

versioned history
https://dnsgjos7lj2fu.cloudfront.net/tests/history/connectors/source-pokeapi/0.1.5.json

connector build report
https://dnsgjos7lj2fu.cloudfront.net/tests/summary/connectors/source-pokeapi/

@evantahler Should I just run the build report on this branch prior to merging?

Comment (evantahler) [reply airbyte-integrations/builds.md:9]: I think you should run the build report immediately after merging (not before, just in case we don't merge this for a while)

Comment (bechurch) [reply airbyte-integrations/builds.md:9]: Added to post merge

**Pull request merged**
